### PR TITLE
add feature authentication with unit test

### PR DESCRIPTION
**Pull request by ftamhar**

### CODE REVIEW

File: common/middlewares/authentication/authentication_test.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	if err != nil {
+		t.Fatal(err)
+	}
+	h.Set("Authorization", token)
+	if status, resp := testRequest(t, ts, "GET", "/", h, nil); status != 200 || resp != "welcome" {
+		t.Fatalf(resp)
+	}
+}
+
+func testRequest(t *testing.T, ts *httptest.Server, method, path string, header http.Header, body io.Reader) (int, string) {
```
Comment (darmawan01) [inline common/middlewares/authentication/authentication_test.go]: Put somewhere that every test can access this, wdyt ?

maybe we can create object for `InitTset()` or `SetupTest()` to help us on test

**Comment by darmawan01** [comment]
Should we use this chi built in ?

https://github.com/go-chi/jwtauth

**Comment by darmawan01** [comment]
And better to tag the ticket if any on commit message, then assign to yourself if you work on it please

Thanks @ftamhar

File: common/middlewares/authentication/authentication_test.go (hunk 1 above)
Comment (ftamhar) [reply common/middlewares/authentication/authentication_test.go]: done at [this](https://github.com/darmawan01/go-grpc-boileplate/pull/10/commits/0fc55ceb65272a163fff3056405b056e060cc8d7) commit

**Comment by ftamhar** [comment]
> Should we use this chi built in ?
> 
> https://github.com/go-chi/jwtauth

If we use that, we are not implement what `RWH` want.

**Comment by ftamhar** [comment]
> And better to tag the ticket if any on commit message, then assign to yourself if you work on it please
> 
> Thanks @ftamhar

No problem.

I don't know how to use tags or ticket on github. :D

**Review by darmawan01 — APPROVED** [review]
LGTM

**Pull request merged**